RIOT-OS/RIOT
# SAUL: Extract name from saul_gpio_params

### Contribution description
For almost all devices in SAUL we currently have a `saul_reg_info_t` structure that holds the information for the SAUL registry and some other structure that holds the parameters needed for its initialization. There are two exceptions to that: GPIOs and ADCs. This difference breaks the pattern and makes it more difficult to treat all devices in a generic way.

This PR extracts the information that actually belongs to a `saul_reg_info_t` structure from the `saul_gpio_params_t` (currently just the name), and creates an array of structures that holds that information, like all other drivers do, making everything more consistent.

Right now moving the names out of the structure might seem a little strange, but there are plans to expand the metadata for the SAUL registry entries, and the place for it is the `saul_reg_info_t` structure.

### Testing procedure
Run `examples/saul` and check that the names of the GPIOs are correctly displayed.

### Issues/PRs references
